src/module.c

```diff
@@ -1262,7 +1262,12 @@ static int DistSearchCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int
   SearchCmdCtx* sCmdCtx = rm_malloc(sizeof(*sCmdCtx));
   sCmdCtx->argv = rm_malloc(sizeof(RedisModuleString*) * argc);
   for (size_t i = 0 ; i < argc ; ++i) {
+#ifdef HAVE_REDISMODULE_HOLDSTRING
     sCmdCtx->argv[i] = RedisModule_HoldString(ctx, argv[i]);
+#else
+    sCmdCtx->argv[i] = argv[i];
+    RedisModule_RetainString(ctx, argv[i]);
```

Comment:
Why not just use retain string?

Reply:
I just wanted something to remind us the right way to go here. We can add a TODO comment if it helps.

Reply:
I guess its also fine :+1:

Reply:
We could use `isFeatureSupported` that we use to identify `RM_Scan`.